internal/app/synchronizer/synchronizer_service.go

```diff
@@ -292,6 +292,11 @@ Registration:
 	<-closedOnCycleEnd
 	stats.Record(ctx, iterationLatency.M(float64(time.Since(cst)/time.Millisecond)))
 
+	err := s.store.CleanupBackfills(ctx)
```

Comment:
for a single returned err, prefer using the more concise pattern: 

if err := s.store.CleanupBackfills(ctx); err != nil {
    logger.Errorf("Failed to clean up backfills, %s", err.Error())
}